spark/src/main/scala/ai/chronon/spark/Analyzer.scala

```diff
@@ -247,11 +250,12 @@ class Analyzer(tableUtils: TableUtils,
            |""".stripMargin)
     }
 
-    val aggMetadata = if (groupByConf.aggregations != null) {
-      groupBy.aggPartWithSchema.map { entry => toAggregationMetadata(entry._1, entry._2) }.toArray
+    val aggMetadata = if (groupByConf.hasDerivations || groupByConf.aggregations == null) {
```

Comment:
@yuli-han 
should it be without the derivations for this condition ?
```suggestion
    val aggMetadata = if (!groupByConf.hasDerivations || groupByConf.aggregations == null) {
```

Reply:
Hi @pengyu-hou it should be with derivation right? Because in derivation case we cannot get the derived feature name from aggregation so have to pull the column name from the output schema.